tracer/src/Datadog.Trace/Util/DomainMetadata.cs

```diff
@@ -18,38 +19,28 @@ internal static class DomainMetadata
         private static string _currentProcessName;
         private static string _currentProcessMachineName;
         private static int _currentProcessId;
+
         private static bool _processDataUnavailable;
         private static bool _domainDataPoisoned;
         private static bool? _isAppInsightsAppDomain;
 
         static DomainMetadata()
         {
             TrySetProcess();
-        }
 
-        public static string ProcessName
-        {
-            get
+            if (_processDataUnavailable)
```

Comment:
I had problems understanding this until I realize `_processDataUnavailable` was set in `TrySetProcess()`. I think it would be easier to understand if we move this condition to set the "unknown" values inside the catch in `TrySetProcess()`.